contracts/Registry.sol

```diff
@@ -142,13 +145,14 @@ contract BeranamesRegistry is
         string[] calldata chars,
         uint duration,
         IERC20 paymentAsset
-    ) external payable {
+    ) external payable validDuration(duration) {
         uint price = priceOracle().price(
             chars,
             duration,
             address(paymentAsset)
         );
         paymentAsset.safeTransferFrom(_msgSender(), address(this), price);
+        paymentAsset.safeApprove(addressesProvider.FUNDS_MANAGER(), 0);
```

Comment:
as we can see in safe erc20 safeApprove works only for initializing value or for resetting. Tbh I checked the allowance before entered this function – allowance was zero but error about it still appeared and only after adding this "resetting" call it started work well